### PR TITLE
[WIP]feat(checkout): CHECKOUT-7506 Design and document v3 management api endpoints contract for custom fee CRUD operations 

### PR DESCRIPTION
**Pull request by bc-shawnwang**
# [DEVDOCS-5078]

## What changed?
Provide a bulleted list in the present tense
* This PR is used to show the changes based on current doc, DO NOT MERGE
* based on our decision for api design: https://bigcommercecloud.atlassian.net/wiki/spaces/CHECK/pages/465765846/Project-5577+Support+order+level+fee+at+the+checkout+via+an+API

## Anything else?
Add related PRs, salient notes, ticket numbers, etc.

ping {names}

[DEVDOCS-5078]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-5078?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/checkouts.v3.yml [hunk 1]
```diff
@@ -8798,6 +8863,28 @@ components:
...
+          enum:
+            - A
+            - B
+        name:
+          type: string
+          description: name of the fee.
+        cost:
+          type: number
+          description: cost of the fee. 
+        tax_class_name:
```
Comment (theromulans) [inline reference/checkouts.v3.yml:8882]:
```suggestion
        tax_class_id:
```

**Comment by valentindellangela** [comment]
No GET? 🤔

**Comment by bc-shawnwang** [comment]
@valentindellangela I thought it's similar to coupon, shipping and item, we don't have a specific endpoint to get the entity of the aggregate? and another api change ticket will include fees on checkout object. how do you think of it?

File: reference/checkouts.v3.yml [hunk 2]
```diff
@@ -8798,6 +8898,31 @@ components:
...
+          description: name of the fee. e.g "Route Package Protection Insurance"
+        display_name:
+          type: string
+          description: display name of the fee targeting customers/shoppers e.g "Package Protection Insurance"
+        cost:
+          type: number
+          description: cost of the fee. 
+        tax_class_id:
+          type: integer
+          description: the tax class id of the fee. 0 => global default tax class id
```
Comment (bc-shawnwang) [inline reference/checkouts.v3.yml:8921]: 0 means global default tax, we need to find the class id in the server side and apply it for tax

Comment (bc-shawnwang) [reply reference/checkouts.v3.yml]: decided to not include tax_class_id atm. we always use default tax class id for fees this milestone (support Route)

**Comment by animesh1987** [comment]
Nice work @bc-shawnwang 👍

**Comment by bc-shawnwang** [comment]
This dev stage document changes are review by checkout team, so no need to be opened. We will create PR to really change the docs after finished our implementation before beta

**Pull request closed without merging**